MarsRover/RoverClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MarsRover
+{
+    public class RoverClient
+    {
+        private Grid _grid;
+        private IHeading _heading;
+        private List<Obstacle> _obstacles;
+        private Rover _rover;
+        private List<RoverNavigationHistory> _navigationHistory = new List<RoverNavigationHistory>();
```

Comment:
Use an `IEnumerable` here please.

Reply:
Also, it appears this value is never used.  If so you can delete it.  I also think you can delete `RoverNavigationHistory` class as well since I can't find any other usages of it.

Reply:
We removed the `RoverNavigationHistory` earlier so we can consider this resolved.